examples/fabric-bridge-app/linux/main.cpp

```diff
@@ -97,10 +101,57 @@ void AttemptRpcClientConnect(System::Layer * systemLayer, void * appState)
 }
 #endif // defined(PW_RPC_FABRIC_BRIDGE_SERVICE) && PW_RPC_FABRIC_BRIDGE_SERVICE
 
+class AdministratorCommissioningCommandHandler : public CommandHandlerInterface
+{
+public:
+    // Register for the AdministratorCommissioning cluster on all endpoints.
+    AdministratorCommissioningCommandHandler() :
+        CommandHandlerInterface(Optional<EndpointId>::Missing(), AdministratorCommissioning::Id)
+    {}
+
+    void InvokeCommand(HandlerContext & handlerContext) override;
+};
+
+void AdministratorCommissioningCommandHandler::InvokeCommand(HandlerContext & handlerContext)
+{
+    using Protocols::InteractionModel::Status;
+
+    EndpointId endpointId = handlerContext.mRequestPath.mEndpointId;
+    ChipLogProgress(NotSpecified, "Received command to open commissioning window on Endpoind: %d", endpointId);
+
+    if (handlerContext.mRequestPath.mCommandId != Commands::OpenCommissioningWindow::Id || endpointId == kRootEndpointId)
+    {
+        // Proceed with default handling in Administrator Commissioning Server
+        return;
+    }
+
+    handlerContext.SetCommandHandled();
+    Status status = Status::Success;
+
+#if defined(PW_RPC_FABRIC_BRIDGE_SERVICE) && PW_RPC_FABRIC_BRIDGE_SERVICE
+    Device * device = DeviceMgr().GetDevice(endpointId);
+    if (device != nullptr && OpenCommissioningWindow(device->GetNodeId()) == CHIP_NO_ERROR)
+    {
+        ChipLogProgress(NotSpecified, "Commissioning window is now open");
```

Comment:
nit, is the commissioning window now open or has the command been sent to open the window?

My understanding is you can call OpenCommissioningWindow with an invalid node id and get a success here because the RPC is sent successfully, but it doesn't imply that a window is actually opened

Reply:
Good catch, if you do not set mDevices[index] back to nullptr when returning -1 due to an error, it can leave the mDevices array in an inconsistent state, where mDevices[index] points to a device that wasn't properly added.

This issue also exist in Bridge Example App Where I copy from, https://github.com/project-chip/connectedhomeip/issues/33946

Reply:
Added comments, we need to make OpenCommissioningWindow synchronous to keep the following logic correct. But I need to integrate this API to CHIP https://pigweed.dev/pw_rpc/#client-synchronous-call-wrappers.

If not feasible, we need to explore other options.